boa_engine/src/context/intrinsics.rs

```diff
@@ -928,3 +950,200 @@ impl IntrinsicObjects {
         self.intl.clone()
     }
 }
+
+#[derive(Debug)]
+pub(crate) struct ObjectTemplates {
```

Comment:
I would add documentation for this structure and methods, since this is something that gets used a lot through the codebase.

This would help future contributors.